Update beginning of the tutorial and revise for recent LAMMPS-GUI changes (#24)
lammpstutorials/lammpstutorials-article: lammps-tutorials.tex

executed \emph{before} the simulation box is created, and they will
cause an error if entered later.  Similarly, many LAMMPS commands may
only be entered \emph{after} the simulation box is defined.  Only a very
small number of commands may be used in both cases.  Edit
\textit{initial.lmp} so that section looks like this:
{\normalsize
\begin{verbatim}
# 1) Initialization
units lj
dimension 3
atom_style atomic
boundary p p p
\end{verbatim}
}

The first line, \textit{units lj}, indicates that we want to use
so-called ``reduced units'' in which all quantities are unitless.  This
is a popular choice for simulations investigating general statistical
mechanical principles where only relative differences between parameters
matter and that are not aimed to represent any specific material.

The second line, \textit{dimension 3}, indicates that the simulation is
in 3D as opposed to a 2D system where atoms can only move in the
xy-plane.  The third line, \textit{atom\_style atomic}, indicates that
the \textit{atomic} style will be used for simple individual particles,
therefore each particle is just a point with a mass.  This is the

> minimal atom style; other atom styles allow to associate more properties
> with atoms like charges, bonds, molecule IDs and much more.  The choice

Reviewer: ```suggestion
minimal atom style; other atom styles allow to associate more properties
with atoms, such as charges, bonds, molecule IDs, and more.  The choice
```

Author: I'm just cutting this down here. The remarks about using a superset may be better given in tutorial 2 where the "external" data file is provided and there the atom style must be compatible etc.

of atom style is generally determined by the model being simulated.

> Using a different atom style is possible for as long as it is a superset

Reviewer: ```suggestion
Using a different atom style is possible as long as it is a superset
```

Author: This changes text deleted in the previous suggestion.

> of the atom style with the required properties.  For example, it would
> be possible to use atom style ``charge'' instead of ``atomic''.  It will
> not affect the simulation, only it requires more memory to store the
> (otherwise unused) per-atom properties.

Reviewer: ```suggestion
of the atom style with the required properties.  For example, you could
use the atom style 'charge' rather than 'atomic'.  This will not affect
the simulation, but it will require more memory to store the (otherwise
unused) per-atom properties.
```

Author: ditto

}

The two \textit{mass} commands assign a mass of 1.0 and 10.0 mass units
to the atoms of type 1 and 2, respectively.

The third line, \textit{pair\_style lj/cut 4.0}, requests that the atoms
will be interacting through a Lennard-Jones (LJ) potential with a
cut-off equal to $r_c = 4.0$ length units
\cite{wang2020lennard,fischer2023history}:
$$E_{ij} (r) = 4 \epsilon_{ij} \left[ \left( \dfrac{\sigma_{ij}}{r} \right)^{12}
  - \left( \dfrac{\sigma_{ij}}{r} \right)^{6} \right], ~ \text{for} ~ r
< r_c,$$ where $r$ is the inter-particle distance, $\epsilon_{ij}$ is
the depth of the potential well that sets the interaction strength, and
$\sigma_{ij}$ represents the distance where the potential energy of the
LJ potential is zero.  Here, the indexes \textit{ij} refer to pairs of
particle types \textit{i} and \textit{j}.  The minimum of the LJ
potential in this formulation is at
$\bar{\sigma} = 2^{\frac{1}{6}}\sigma$.  Sometimes people use an
alternate formulation of the LJ potential which uses $\bar{\sigma}$,
i.e.~the minimum of the potential.  By substituting
$\sigma = \frac{\bar{\sigma}}{2^{\frac{1}{6}}}$ in the formulation above
we get the alternate but equivalent expression:

> $$E_{ij} (r) = \epsilon_{ij} \left[ \left( \dfrac{\bar{\sigma}_{ij}}{r} \right)^{12}

Reviewer: I am not convinced that the part about the alternate formulation of the LJ potential is necessary for absolute LAMMPS beginner facing their first tutorial (unless you've noticed that this particular confusion was commonplace?). In the website, I place such "extra" information in color boxes that users can skip if they don't want the info, this piece of information could definitely go there, but for the articles I think we could go "straight to business". 

edit: may be to be placed in a "\begin{note}" block?

After reading the "pair_style lj/cut" page, I saw a few typo. I remmeber that you sometimes collect "typo" from the Matsi forum in the past and fix them yourself, but to give you less work, can I simply create a PR with such minor corrections on the lammps github? (I am aware that the LAMMPS documentation does not encourage individual PR for minor correction though : "If you find an error or omission in the LAMMPS documentation or on the LAMMPS website, or have a suggestion or content for something to clarify or include, send an email to the developers." from https://www.lammps.org/contribute.html)

Author: > I am not convinced that the part about the alternate formulation of the LJ potential is necessary for absolute LAMMPS beginner facing their first tutorial

I am tempted to mention this, because the question of which pair style to use for the alternate form is a recurring one. I just noticed that the same information, perhaps a bit more terse is in the lj/cut pair style documentation at a rather prominent place. So it should be reduced to a short statement, perhaps in a "note" box. It is particularly beginners, that confuse there. So it cannot be mentioned enough. At the same time, expanding the LAMMPS manual and just pointing to it, may be just as well.

> can I simply create a PR with such minor corrections on the lammps github
You can submit a "collection" PR like I often do. Just call it a collection of small fixes for the manual and submit it as Draft. ... or just add review suggestions like you are doing here to my current collection (which is growing beyond proportion due to the LAMMPS-GUI changes and the fact that nobody is approving it, so I am adding to it and not merging it. I expect more updates for the manual soon because at the developer meeting tomorrow, I would like to organize a general pre-stable documentation check across the developers where we may find more things. We're still about 10 days away from the planned release day.

thus have: $\epsilon_{12} = \sqrt{1.0 \times 0.5} = 0.707$, and
$\sigma_{12} = \sqrt{1.0 \times 3.0} = 1.732$.

\paragraph{Single point energy}
The system is now fully parameterized and the input ready to compute
forces.  Let us fill in the two last remaining categories,
\textit{Visualization}, and \textit{Run}, by adding the following lines
into \textit{initial.lmp}:
{\normalsize
\begin{verbatim}
# 4) Visualization
thermo 10
thermo_style custom step etotal press

# 5) Run
run 0 post no
\end{verbatim}
}
The \textit{thermo} command asks LAMMPS to print thermodynamic

> information to the console every given number of steps, in this input
> every 10 simulation steps.  The \textit{thermo\_style custom} determines

Reviewer: ```suggestion
information to the console every given number of steps, in this case,
every 10 simulation steps.  The \textit{thermo\_style custom} command specifies
```

\emph{which} output LAMMPS should print.  This selection prints the the
step number (\textit{step}), the total energy (\textit{etotal}), and the

> pressure (\textit{press}).  The \textit{run 0} command will tell LAMMPS
> just to a force and energy initialization.

Reviewer: ```suggestion
pressure (\textit{press}).  The \textit{run 0} command instructs LAMMPS
to perform only force and energy initialization.
```

You can now run LAMMPS.  It should finish quickly and with the default
settings LAMMPS--GUI will open two new windows, one with the console
output and one with a chart.  The output window will show some info
printed by the various command in the input as they are executed and

> the total energy and pressure at step 0, the requested thermodynamic
> data.  Since we have not done any real simulation steps, the charts

Reviewer: ```suggestion
the total energy and pressure at step 0, i.e. the requested thermodynamic
data.  Since no real simulation steps have been performed, the charts
```

are empty.

\begin{figure}
\centering
\includegraphics[width=0.55\linewidth]{LJ}
\caption{The binary mixture simulated during \hyperref[lennard-jones-label]{Tutorial 1}.
  The atoms of type 1 are represented as small red spheres, the atoms of type 2 as large
  green spheres, and the edges of the simulation box are represented a blue sticks}
\label{fig:LJ}
\end{figure}

\paragraph{Snapshot Image}

At this point it is also possible to create a snapshot image of the
current system using the \textit{Image Viewer} window.  The image viewer
works by telling LAMMPS to render an image of the current system using
its own render library through the \textit{dump image} command.  The

> resulting is read in and displayed.  The various buttons can be used to

Reviewer: the resulting image ?

Author: yes.

\label{fig:chart-log}
\end{figure}

This tells LAMMPS to perform an energy minimization of the system.  That
is, LAMMPS will compute the forces on all atoms and then update the
positions according to a selected algorithm so that the potential energy
will decrease.  By default, LAMMPS uses the conjugate gradient (CG)
algorithm \cite{hestenes1952methods}.  The simulation will stop as soon
at the minimizer algorithm cannot find a way to to lower the potential

> energy.  Except for trivial systems, minimization algorithms will find
> the ``next'' minimum, which may not be the global minimum.

Reviewer: ```suggestion
energy.  Except for trivial systems, minimization algorithms will find
a local minimum rather than the global minimum.
```

Run the minimization and observe that LAMMPS-GUI will capture the output
and update the chart in realtime.  This run executes fast (depending on
the capability of your computer, of course), and thus may fail to
capture some of the thermodynamic data (see figure \ref{fig:chart-log}).
In that case, you can use the \textit{Preferences} dialog to reduce the
GUI update interval and switch to single threaded, unaccelerated

> execution in the Accelerators tab.  You can repeat the run.  A new run
> will always ``forget'' the current system and start over from the top.

Reviewer: ```suggestion
execution in the Accelerators tab.  You can repeat the run.  Each new run
will start fresh, forgetting the current system and starting over from the top.
```

tutorial setup.  This contains the \textit{Part A} of the
\textit{initial.lmp} file \emph{without} the \textit{create\_atoms}
commands in the \textit{System definition} part.

We want to create the the atoms of types 1 and 2 in two separate
regions.  Thus we need to add two \textit{region} commands and then
re-add \textit{create\_atoms} commands, but using the new regions
instead of the simulation box region for placing the created atoms:
{\normalsize
\begin{verbatim}
# 2) System definition
region simulation_box block &
     -20 20 -20 20 -20 20
create_box 2 simulation_box
# for creating atoms
region cylinder_in cylinder z 0 0 10 &
     INF INF side in
region cylinder_out cylinder z 0 0 10 &
     INF INF side out
create_atoms 1 random 1000 341341 cylinder_out
create_atoms 2 random 150 127569 cylinder_in
\end{verbatim}
}
The \textit{side in} and \textit{side out} keywords are used to define
regions that represent the inside and outside of the cylinder of radius
10. Then append a sixth section \textit{Save system} at the end of the
file with the following content.:
{\normalsize
\begin{verbatim}
# 6) Save system
write_data minimized_coordinates.data
\end{verbatim}
}

> The significant improvement input is the addition of the

Reviewer: ```suggestion
A significant improvement to the input is the addition of the
```

\textit{write\_data} command.  This command is used to write the state
of the system a text format file called \textit{improved.min.data}.

> Note that the \textit{write\_data} command is placed \emph{after} the

Reviewer: ```suggestion
Make sure the \textit{write\_data} command is placed \emph{after} the
```
